plaidml/op/op.h

```diff
@@ -725,5 +725,49 @@ inline edsl::Tensor variance(const edsl::Tensor& I, const edsl::Value& axes = ed
   return details::op("variance", args).as_tensor();
 }
 
+struct OpTensor {
+  edsl::Tensor tensor;
+  OpTensor(convolution op) : tensor(edsl::Tensor(op)) {}       // NOLINT
+  OpTensor(explicit_padding op) : tensor(edsl::Tensor(op)) {}  // NOLINT
+  OpTensor(lrn op) : tensor(edsl::Tensor(op)) {}               // NOLINT
+  OpTensor(mvn op) : tensor(edsl::Tensor(op)) {}               // NOLINT
+  OpTensor(l2norm op) : tensor(edsl::Tensor(op)) {}            // NOLINT
+  OpTensor(relu op) : tensor(edsl::Tensor(op)) {}              // NOLINT
+  OpTensor(repeat op) : tensor(edsl::Tensor(op)) {}            // NOLINT
+  OpTensor(slice op) : tensor(edsl::Tensor(op)) {}             // NOLINT
+};
+
+#define PLAIDML_OPLIB_OP_CONVERSION(_op_)                                                               \
```

Comment:
Why do we need to duplicate all of these?

Reply:
So, when I modified the operator overloads replacing the `Tensor` inputs with `LocatedTensor`, there were broader ramifications than I originally realized. All those operators work fine for `Tensor` operands, which now get implicitly converted to `LocatedTensors`, but there were a number of non-Tensor classes that were relying on those overloads to be cast into `Tensors`, and with the change to `LocatedTensors` no longer will be; ie, the compiler does not automatically convert them to `Tensors` then to `LocatedTensors`. One example of this is the `Contraction` class, which I was able to handle pretty easily by just adding a `Contraction` constructor to `LocatedTensor`. The real issue is for all of the classes defined in `op.h`; since `op.h` depends on `edsl.h`, I couldn't just include `op.h` and add the necessary constructors to `LocatedTensor` without creating a circular dependency. 

I'm not a huge fan of my fix here, but it was the best I could come up with. Other options I thought of would be to individually define all of the operators for each `op.h` class, which made the code messier without really offering any benefit to this approach, or to pull the `LocatedTensor` definition and associated operators into its own header file, which might be a bit neater, but I thought would lead to some awkward inclusion requirements (ie, anywhere `edsl.h` or `op.h` is included, after those inclusions). The ideal fix would be to make the relevant `op.h` classes available to `edsl.h` somehow but I couldn't figure out a way to do this.